### PR TITLE
Fix inconstancy in the login and logout consent flows

### PR DESCRIPTION
**Pull request by senthalan**
### Proposed changes in this pull request
This will introduce the following two new properties in the identity.xml inside OpenIDConnect tag,
1. SkipLoginConsent - Skip the OIDC login consent
2. SkipLogoutConsent - Skip the OIDC logout consent

If these properties are not defined, then it will be retrieved from SkipUserConsent

Fix https://github.com/wso2/product-is/issues/7256

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java [hunk 1]
```diff
@@ -1167,12 +1169,33 @@ public String getOpenIDConnectIDTokenSubjectClaim() {
     }
 
     /**
-     * Returns if skip user consent enabled or not
+     * Returns if login consent enabled or not
```
Comment (Buddhimah) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: shall we add a full stop at the end

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java [hunk 2]
```diff
@@ -1167,12 +1169,33 @@ public String getOpenIDConnectIDTokenSubjectClaim() {
...
      *
-     * @return
      */
     public boolean getOpenIDConnectSkipeUserConsentConfig() {
-        return "true".equalsIgnoreCase(openIDConnectSkipUserConsent);
+
+        if (openIDConnectSkipLoginConsent == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("The SkipLoginConsent property is not configured. So retrieving the SkipUserConsent value");
```
Comment (Buddhimah) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: shall we add a full stop at the end

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java [hunk 3]
```diff
@@ -1167,12 +1169,33 @@ public String getOpenIDConnectIDTokenSubjectClaim() {
...
+            }
+            Boolean.parseBoolean(openIDConnectSkipUserConsent);
```
Comment (Buddhimah) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: why we are using `openIDConnectSkipUserConsent `variable as a string ?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java [hunk 4]
```diff
@@ -2510,12 +2520,19 @@ private void parseOpenIDConnectConfig(OMElement oauthConfigElem) {
...
-                openIDConnectSkipUserConsent =
+                openIDConnectSkipLoginConsent = Boolean.parseBoolean(
                         openIDConnectConfigElem.getFirstChildWithName(
-                                getQNameWithIdentityNS(ConfigElements.OPENID_CONNECT_SKIP_USER_CONSENT))
-                                .getText().trim();
+                                getQNameWithIdentityNS(ConfigElements.OPENID_CONNECT_SKIP_LOGIN_CONSENT))
+                                .getText().trim());
+            }
+            if (openIDConnectConfigElem.getFirstChildWithName(
+                    getQNameWithIdentityNS(ConfigElements.OPENID_CONNECT_SKIP_LOGOUT_CONSENT)) != null) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: Shouldn't this be backward compatible? That means if 
(ConfigElements.OPENID_CONNECT_SKIP_LOGOUT_CONSENT)) is null, shouldn't we read ConfigElements.OPENID_CONNECT_SKIP_USER_CONSENT from config file and set its value to 'openIDConnectSkipLogoutConsent'?

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 5]
```diff
@@ -752,14 +752,23 @@ private void removeSessionDataFromCache(String sessionDataKey) {
      */
     private boolean getOpenIDConnectSkipUserConsent(String idTokenHint) throws ParseException, IdentityOAuth2Exception {
 
+        boolean skipLogoutConsentConfig =
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: shall we rename the variable name to skipLogoutConsent?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java (hunk 4 above)
Comment (senthalan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: I have added the old configuration as a fallback

**Pull request merged**